Date: Fri, 28 Jul 2023 05:49:53 +0000
Subject: Fix the issue causing Chromium to hang at audio_thread_->Stop during AttemptRestart
Signed-off-by: Fangzhou Chen <fangzhou@fydeos.io>

Generated by `git diff 4cb0c5189686d5a55f37f222c53127b160d10686 0c347f406fa7af8d2c2cd32d074a716359c960f0`
in `src/third_party/adhd`

`0c347f406fa7af8d2c2cd32d074a716359c960f0` is `CROS_WORKON_COMMIT` in adhd-0.0.7-r3006.ebuild

`4cb0c5189686d5a55f37f222c53127b160d10686` is new commit https://chromium-review.googlesource.com/c/chromiumos/third_party/adhd/+/4536936

After the above commit is included in the updated ebuild CROS_WORKON_COMMIT, this patch is no longer required.

diff --git a/cras/src/server/cras_iodev_list.c b/cras/src/server/cras_iodev_list.c
index 2a5edad7..244bb0bb 100644
--- a/cras/src/server/cras_iodev_list.c
+++ b/cras/src/server/cras_iodev_list.c
@@ -371,8 +371,7 @@ static int fill_node_list(struct iodev_list* list,
 
   const bool dsp_nc_allowed = !get_nc_blocked_state() ||
                               cras_system_get_bypass_block_noise_cancellation();
-  const bool ap_nc_allowed =
-      cras_feature_enabled(CrOSLateBootAudioAPNoiseCancellation);
+  const bool ap_nc_allowed = false;
 
   DL_FOREACH (list->iodevs, dev) {
     DL_FOREACH (dev->nodes, node) {
diff --git a/cras/src/server/cras_processor_config.c b/cras/src/server/cras_processor_config.c
index 144641e7..47cc558b 100644
--- a/cras/src/server/cras_processor_config.c
+++ b/cras/src/server/cras_processor_config.c
@@ -12,8 +12,7 @@
 
 enum CrasProcessorEffect cras_processor_get_effect(bool nc_provided_by_ap) {
   if (nc_provided_by_ap && cras_system_get_noise_cancellation_enabled() &&
-      cras_system_get_ap_noise_cancellation_supported() &&
-      cras_feature_enabled(CrOSLateBootAudioAPNoiseCancellation)) {
+      cras_system_get_ap_noise_cancellation_supported() && false) {
     return NoiseCancellation;
   }
   return NoEffects;
diff --git a/cras/src/server/rust/cras_dlc/src/lib.rs b/cras/src/server/rust/cras_dlc/src/lib.rs
index e78530ea..4b5353dc 100644
--- a/cras/src/server/rust/cras_dlc/src/lib.rs
+++ b/cras/src/server/rust/cras_dlc/src/lib.rs
@@ -17,7 +17,7 @@ use system_api::dlcservice::DlcState_State;
 use system_api::dlcservice::InstallRequest;
 use thiserror::Error;
 
-const DBUS_TIMEOUT: Duration = Duration::from_secs(3);
+const DBUS_TIMEOUT: Duration = Duration::from_millis(500);
 
 #[derive(Error, Debug)]
 enum Error {
